File: rb130/intro_to_regex_book/using_regex_in_ruby_and_javascript/ex1.rb
def url?(str)
  str.match?(/\Ahttps?:\/\/\S+\z/)
end

p url?('http://launchschool.com')   # => true
p url?('https://example.com')       # => true
p url?('https://example.com hello') # => false
p url?('   https://example.com')    # => false

# ALTERNATIVE SOLUTION:

# def url?(text)
#   !!text.match(/\Ahttps?:\/\/\S+\z/)
# end

File: intro_to_regex_book/using_regex_in_ruby_and_javascript/ex1.js
function isUrl(text) {
  let urlRegex = /^https?:\/\/\S+$/
  return urlRegex.test(text);
}

console.log(isUrl('http://launchschool.com'));   // -> true
console.log(isUrl('https://example.com'));       // -> true
console.log(isUrl('https://example.com hello')); // -> false
console.log(isUrl('   https://example.com'));    // -> false


// ALTERNATIVE SOLUTION:

// let isUrl = function (text) {
//   return !!text.match(/^https?:\/\/\S+$/);
// };
